### PR TITLE
chore: add logger package deploy

### PR DESCRIPTION
**Pull request by AustinAbro321**
## Description

add slogger for package deploy. Example below. Note that we will not print connect strings or the credentials table after deploy for `--log-format="console"` after this is merged. Once we've refactored to packager2 it becomes easier to do that in the future. For now, since we have the commands `zarf connect list` and `zarf tools get-creds` which achieve the same affect, I feel comfortable with this change.   
image

## Related Issue

Relates to #2576 

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/zarf-dev/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/pkg/packager/deploy.go [hunk 1]
```diff
@@ -89,12 +92,18 @@ func (p *Packager) Deploy(ctx context.Context) error {
 		return err
 	}
 	warnings = append(warnings, validateWarnings...)
+	for _, warning := range validateWarnings {
```
Comment (mkcp) [inline src/pkg/packager/deploy.go:95]: note: any time I see a simple "take collection, apply function to each" I make this face 😃

File: src/pkg/packager/deploy.go [hunk 2]
```diff
@@ -124,10 +133,12 @@ func (p *Packager) Deploy(ctx context.Context) error {
 	}
 	if len(deployedComponents) == 0 {
 		message.Warn("No components were selected for deployment.  Inspect the package to view the available components and select components interactively or by name with \"--components\"")
+		l.Warn("No components were selected for deployment.  Inspect the package to view the available components and select components interactively or by name with \"--components\"")
```
Comment (mkcp) [inline src/pkg/packager/deploy.go:136]: nit: capitalization. admittedly this one is weird cause of the second sentence. I was doing lower first sentence, upper second.

File: src/pkg/packager/deploy.go [hunk 3]
```diff
@@ -124,10 +133,12 @@ func (p *Packager) Deploy(ctx context.Context) error {
...
 	}
 
 	// Notify all the things about the successful deployment
 	message.Successf("Zarf deployment complete")
+	l.Debug("Zarf deployment complete", "duration", time.Since(start))
```
Comment (mkcp) [inline src/pkg/packager/deploy.go:141]: Nit: capital. 👍 on the duration metric

File: src/pkg/packager/deploy.go [hunk 4]
```diff
@@ -357,6 +378,7 @@ func (p *Packager) deployComponent(ctx context.Context, component v1alpha1.ZarfC
 		healthCheckContext, cancel := context.WithTimeout(ctx, p.cfg.DeployOpts.Timeout)
 		defer cancel()
 		spinner := message.NewProgressSpinner("Running health checks")
+		l.Info("running health checks")
```
Comment (mkcp) [inline src/pkg/packager/deploy.go:381]: Non-blocking: Is there a piece of data we can use to link this to the component e.g. component name? If we grow this part of the system to be concurrent and healthchecks are interleaved together, we want the ordering to not matter for the logs to make sense.

File: src/pkg/packager/deploy.go [hunk 5]
```diff
@@ -368,16 +390,22 @@ func (p *Packager) deployComponent(ctx context.Context, component v1alpha1.ZarfC
...
 	}
+	l.Debug("done deploying component", "name", component.Name, "duration", time.Since(start))
 	return charts, nil
 }
 
 // Move files onto the host of the machine performing the deployment.
-func (p *Packager) processComponentFiles(component v1alpha1.ZarfComponent, pkgLocation string) error {
+func (p *Packager) processComponentFiles(ctx context.Context, component v1alpha1.ZarfComponent, pkgLocation string) error {
+	l := logger.From(ctx)
+	l.Handler()
```
Comment (mkcp) [inline src/pkg/packager/deploy.go:400]: Bug: Is this intentional? It doesn't look like this value is used.

File: src/pkg/packager/deploy.go [hunk 6]
```diff
@@ -475,6 +509,7 @@ func (p *Packager) setupState(ctx context.Context) error {
 
 		// Try to create the zarf namespace
 		spinner.Updatef("Creating the Zarf namespace")
+		l.Info("creating the Zarf namespace")
```
Comment (mkcp) [inline src/pkg/packager/deploy.go:512]: Non-blocking: Can we easily add the namespace name here?

File: src/pkg/packager/sources/tarball.go [hunk 7]
```diff
@@ -100,12 +105,14 @@ func (s *TarballSource) LoadPackage(ctx context.Context, dst *layout.PackagePath
 	if !dst.IsLegacyLayout() {
 		spinner := message.NewProgressSpinner("Validating full package checksums")
 		defer spinner.Stop()
+		l.Info("validating package checksums")
```
Comment (mkcp) [inline src/pkg/packager/sources/tarball.go:108]: Non-blocking: Is there an easy piece of metadata we can attach this to? Package name or source maybe?

File: src/pkg/zoci/common.go [hunk 8]
```diff
@@ -29,12 +31,15 @@ type Remote struct {
 
 // NewRemote returns an oras remote repository client and context for the given url
 // with zarf opination embedded
-func NewRemote(url string, platform ocispec.Platform, mods ...oci.Modifier) (*Remote, error) {
-	logger := slog.New(message.ZarfHandler{})
+func NewRemote(ctx context.Context, url string, platform ocispec.Platform, mods ...oci.Modifier) (*Remote, error) {
+	l := slog.New(message.ZarfHandler{})
+	if logger.Enabled(ctx) {
```
Comment (mkcp) [inline src/pkg/zoci/common.go:36]: After seeing this fn used a few times, this looks like an effective way to propagate feature flags through the system. Imo we should look to use this pattern in the future.

**Review by mkcp — CHANGES_REQUESTED** [review]
A few nits around log captialization, some non-blocking comments about adding metadata to some log lines, and a potential bug/dead code about a `Handler()` value that doesn't seem to be used. Overall, good work!

File: src/pkg/packager/deploy.go (hunk 6 above)
Comment (AustinAbro321) [reply src/pkg/packager/deploy.go:512]: We're creating thee Zarf namespace here - the namespace called Zarf

File: src/pkg/packager/deploy.go (hunk 3 above)
Comment (AustinAbro321) [reply src/pkg/packager/deploy.go:141]: This one was on purpose. I've been capitalizing proper nouns like Zarf and Helm. I'm fine with either choice, lmk what you'd like to do going forward, as I currently have the word Helm capitalized a few times in this PR.

File: src/pkg/packager/deploy.go (hunk 5 above)
Comment (AustinAbro321) [reply src/pkg/packager/deploy.go]: Totally missed that good catch

File: src/pkg/packager/deploy.go (hunk 4 above)
Comment (AustinAbro321) [reply src/pkg/packager/deploy.go:381]: Yup added the chart name

**Review by mkcp — APPROVED** [review]

**Pull request merged**